File: maple-core/src/main/scala/org/xi/maple/api/MapleSource.scala
package org.xi.maple.api

import org.apache.spark.sql.{Dataset, Row, SparkSession}
import org.xi.maple.model.SourceConfig

trait MapleSource[T <: SourceConfig] extends MaplePlugin[T] with Logging {
  def getData(spark: SparkSession): Dataset[Row]
}

File: maple-engine-plugins/maple-flink/apps/maple-flink-data-calc/src/main/java/org/xi/maple/datacalc/flink/api/MapleSource.java
package org.xi.maple.datacalc.flink.api;

import org.apache.flink.table.api.TableDescriptor;
import org.apache.flink.table.api.TableEnvironment;
import org.xi.maple.datacalc.flink.model.SourceConfig;
import org.xi.maple.datacalc.flink.util.TableUtils;

public abstract class MapleSource<T extends SourceConfig> extends MaplePlugin<T> implements TableDefine {

    public MapleSource(TableEnvironment tableEnv) {
        super(tableEnv);
    }

    @Override
    public void define() {
        TableDescriptor tableDescriptor = TableUtils.getTableDescriptor(config);
        tableEnv.createTemporaryTable(config.getResultTable(), tableDescriptor);
    }
}
